src/utils/multiSig/getEligibleSignees.ts

```diff
@@ -0,0 +1,142 @@
+  requiredRoles,
+}: GetEligibleSigneesParams): Promise<GetEligibleSigneesResult> {
+  const domainIds =
+    domainId === Id.RootDomain ? [Id.RootDomain] : [Id.RootDomain, domainId];
+  const signeesPerRole: SigneesPerRole = {};
+  const uniqueEligibleSignees: SigneeEntry = {};
+
+  const roleFilter = requiredRoles
+    .filter((role) => !!COLONY_ROLE_TO_FILTER_KEY[role])
+    .map((role) => ({ [COLONY_ROLE_TO_FILTER_KEY[role]]: { eq: true } }));
+
+  const queryFilter: ModelColonyRoleFilterInput = {
+    isMultiSig: { eq: true },
+    or: roleFilter,
+  };
+  // amplify doesn't provide it in the UI, but the syntax of or: [{att: {eq: SOMETHING}}, {att2: {eq: SOMETHING@}}] works!
+  // but sadly having 2 or statements is impossible so we need to loop through domains
+
+  await Promise.all(
+    domainIds.map(async (permissionDomainId) => {
+      const response = await apolloClient.query<
+        GetMultiSigRolesForDomainQuery,
+        GetMultiSigRolesForDomainQueryVariables
+      >({
+        query: GetMultiSigRolesForDomainDocument,
+        variables: {
+          domainId: `${colonyAddress}_${permissionDomainId}`,
+          filter: queryFilter,
+        },
+        fetchPolicy: 'cache-first',
+      });
+
+      const usersWithRoles: ColonyUserRoleFragment[] = (
+        response.data.getRoleByDomainAndColony?.items || []
+      ).filter(notMaybe);
+
+      usersWithRoles.forEach((userRoleEntry) => {
+        if (!notMaybe(userRoleEntry.targetUser)) {
+          return;
+        }
+
+        const userAddress = userRoleEntry.targetAddress;
+
+        // first we determine all the roles user is holding
+        const userRoles = requiredRoles.filter(
+          (role) => !!userRoleEntry[COLONY_ROLE_TO_FILTER_KEY[role]],
+        );
+
+        // This means a user has the same permissions in both the parent and the child domain
+        // The main edge case here is: a user has Administration in root and Funding in the child domain, the action is simple payment
+        // in that case we can just concat the array since they will sign for both roles
+        const existingUserEntry = uniqueEligibleSignees[userAddress];
+
+        if (existingUserEntry) {
+          uniqueEligibleSignees[userAddress] = {
+            ...existingUserEntry,
+            userRoles: [
+              ...new Set([...existingUserEntry.userRoles, ...userRoles]),
+            ],
+          };
+        } else {
+          uniqueEligibleSignees[userAddress] = {
+            userAddress,
+            user: userRoleEntry.targetUser,
+            userRoles,
+          };
+        }
+
+        userRoles.forEach((role) => {
+          if (!signeesPerRole[role]) {
+            signeesPerRole[role] = {};
+          }
+
+          signeesPerRole[role][userAddress] = {
+            userAddress,
+            // For some reason TS doesn't want to follow through with the type assumption here since it's a nested loop
+            user: userRoleEntry.targetUser as Partial<User>,
```

Comment:
if Partial<User> works, it's probably because there's some nullable field that `user` expects to always have a value 🤷‍♂️